ttnn/cpp/ttnn/tensor/serialization.hpp

```diff
@@ -9,16 +9,18 @@
 #include <string>
 #include <unordered_map>
 
-namespace tt {
-
-namespace tt_metal {
+namespace tt::tt_metal {
 
 void dump_tensor(
     const std::string& file_name, const Tensor& tensor, const std::unordered_map<std::string, std::string>& strategy);
 
 Tensor load_tensor(const std::string& file_name, Device* device = nullptr);
 Tensor load_tensor(const std::string& file_name, distributed::MeshDevice* device = nullptr);
 
-}  // namespace tt_metal
+void dump_memory_config(std::ostream& output_stream, const MemoryConfig& memory_config);
```

Comment:
should rename to `save` to mirror `load`?

Reply:
We have some public-visible APIs that rely on _dump_... (reference models / jupyter notebooks). I think it might be a good idea, but probably not in this PR? Just wanted to make `types` smaller:)